lib/Cargo.toml

```diff
@@ -19,7 +19,7 @@ log = "0.4.14"
 # For example programs
 [dev-dependencies]
 ctrlc = "3.1.4"
-clap = "3.2"
+clap-3 = { package = "clap", version = "3.2" }
```

Comment:
So IIUC the point of this commit is to allow the next one to use clap 4 for the new example it introduces. I don't really understand why we cannot instead do one of the following:

* update the whole project to use clap 4, or
* use clap 3 with the new example.

I'd question the usefulness of the new example anyway, since it just does a get/set on a control and requires a FWHT stateless device. I'd say it's probably better to remove it for now, and we will introduce a stateless FWHT decoder later.

Reply:
IMO clap 4 is so much more concise and provides definitely enough support for simple use cases. I agree that moving the whole project to clap 4 would be good. But at the same time I thought that substantially changing a number of otherwise unrelated files would be too much of a change - I wanted to minimize the changes. My reasoning was then to allow the two side-by-side and then (maybe) gradually phase out clap 3.
 
 That said, I'm fine with omitting the example programs for now.

Reply:
Yep, if you can reduce this PR to 3 commits and remove the introduction of clap 4 I think this should be mergeable - then I'll address remaining concerns with follow-up CLs (as I have to think a bit more about it and it would be nice to move forward meanwhile).

Reply:
Pushed the 3 commits